src/core/Table/Table.tsx

```diff
@@ -558,13 +558,13 @@ export const Table = <
   const bodyRef = React.useRef<HTMLDivElement>(null);
 
   const getPreparedRow = React.useCallback(
-    (row: Row<T>) => {
+    (row: Row<T>, index: number) => {
       prepareRow(row);
       return (
         <TableRowMemoized
           row={row}
           rowProps={rowProps}
-          isLast={row.index === data.length - 1}
+          isLast={index === page.length - 1}
```

Comment:
I was looking at solutions for this yesterday, and I just changed `data.length` to `instance.rows.length` to solve the issue

Reply:
I am afraid that `instance.rows` might be not in sync with `page` and we are using `page` to construct Table body rows.